### PR TITLE
Add terraform command output validations

### PR DESCRIPTION
**Pull request by mouellet**
## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------

## What is the new behavior?
-------------------------------------

Adds the ability to test custom variable validations rules (introduced in Terraform v0.13) w/ user-defined assertions on the terraform command output. 

## Does this introduce a breaking change?
-------------------------------------
- [NO]

## Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

## Other information
-------------------------------------

### CODE REVIEW

File: unit/unit.go [hunk 1]
```diff
@@ -58,13 +62,26 @@ func RunUnitTests(fixture *UnitTestFixture) {
...
 	tfPlanFilePath := filepath.FromSlash(fmt.Sprintf("%s/%s.plan", os.TempDir(), random.UniqueId()))
-	terraform.RunTerraformCommand(
+	defer os.Remove(tfPlanFilePath)
+
+	output, err := terraform.RunTerraformCommandE(
 		fixture.GoTest,
 		fixture.TfOptions,
 		terraform.FormatArgs(fixture.TfOptions, "plan", "-input=false", "-out", tfPlanFilePath)...)
-	defer os.Remove(tfPlanFilePath)
+	if err != nil {
```
Comment (nmiodice) [inline unit/unit.go]: I think we have some edge cases here:

Ideally we want:
 * `err == nil`, `CommandStdoutAssertions == nil` --> run `validateTerraformPlanFile`
 * `err != nil`, `CommandStdoutAssertions == nil` --> fail
 * `err == nil`, `CommandStdoutAssertions != nil` --> fail
 * `err != nil`, `CommandStdoutAssertions != nil` --> run `validateTerraformCommandStdout`
 
 This implementation does the following, though:
 * `err == nil` --> run `validateTerraformPlanFile`. If `CommandStdoutAssertions != nil`, these assertions are ignored
 * `err != nil` --> run `validateTerraformCommandStdout`. If `CommandStdoutAssertions == nil` then we always pass

Comment (nmiodice) [reply unit/unit.go]: Can we improve the conditions checking here to cover all cases correctly?

**Review by nmiodice — CHANGES_REQUESTED** [review]
Love the addition! One comment, but otherwise I think the addition is very much welcome :)

File: unit/unit.go (hunk 1 above)
Comment (mouellet) [reply unit/unit.go]: Your right there are some cases missing!

However, for this one:
> * `err == nil`, `CommandStdoutAssertions != nil` --> fail

It think it should not fail but run both `validateTerraformPlanFile` & `validateTerraformCommandStdout`. Someone might want to test the command output even if it didn't failed. Checking for warnings or deprecation notice maybe?

What do you think?

Comment (nmiodice) [reply unit/unit.go]: That's a good point, and I agree!

**Review by nmiodice — APPROVED** [review]
Ship It!
```
                __/___
          _____/______|
  _______/_____\_______\_____
  \              < < <       |
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by nmiodice** [comment]
This LGTM. Can you resolve the conflicts between your two commits and re-push? I'll also be merging a commit shortly so a re-base would be helpful.

**Comment by nmiodice** [comment]
Bonus points if you add a unit test (see latest commit in `master`) to cover this scenario, but it is not necessary

**Review by nmiodice — APPROVED** [review]

**Pull request merged**